### PR TITLE
Stream responsibility

### LINKED ISSUE #51: Assigning stream responsibility
- First node asks tracker who is responsible for stream.
- If no one is responsible, then tracker assigns responsibility to this node.

### PR DESCRIPTION
**Pull request by mirpo**
fixes #51

### CODE REVIEW

File: src/logic/Node.js [hunk 1]
```diff
@@ -52,32 +46,44 @@ class Node extends EventEmitter {
...
+    isOwnStream(streamId) {
+        return this.ownStreams.has(streamId)
     }
 
     _isKnownStream(streamId) {
         return this.knownStreams.get(streamId) !== undefined
     }
+
+    stop(cb) {
+        this.protocols.nodeToNode.stop(cb)
```
Comment (harbu) [inline src/logic/Node.js:85]: Should we stop `trackerNode` also?

File: src/protocol/TrackerNode.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ const events = Object.freeze({
     CONNECTED_TO_TRACKER: 'streamr:peer:send-status',
     NODE_LIST_RECEIVED: 'streamr:node-node:connect',
     DATA_RECEIVED: 'streamr:node-node:stream-data',
-    STREAM_INFO_RECEIVED: 'streamr:node:found-stream'
+    STREAM_INFO_RECEIVED: 'streamr:node:found-stream',
+    STREAM_INFO_OWN_RECEIVED: 'streamr:node:found-own-stream',
```
Comment (harbu) [inline src/protocol/TrackerNode.js]: Hmmm.. what about name `STREAM_ASSIGNED`?

**Review by harbu — COMMENTED** [review]
Looks good to me. Few comments; but nothing critical. Feel free to merge if you want.

File: src/logic/Node.js (hunk 1 above)
Comment (mirpo) [reply src/logic/Node.js:85]: hmmm, we are using one connection, so we don't need to stop all protocols, we only need to stop libp2p connection

File: src/protocol/TrackerNode.js (hunk 2 above)
Comment (mirpo) [reply src/protocol/TrackerNode.js]: good point

**Pull request merged**